src/dist/manifestation.rs

```diff
@@ -242,15 +216,41 @@ impl Manifestation {
             tx = self.uninstall_component(component, new_manifest, tx, download_cfg.process)?;
         }
 
-        // Install components
-        for (component_bin, installer_file) in things_to_install {
-            tx = component_bin.install(installer_file, tx, self)?;
+        let mut downloads = FuturesUnordered::new();
+        let mut component_iter = components.iter();
+        let mut cleanup_downloads = vec![];
+        loop {
+            if downloads.is_empty() && component_iter.len() == 0 {
+                break;
+            }
+
+            let (bin, downloaded) = match downloads.next().await {
+                Some(Ok((bin, downloaded))) => (bin, downloaded),
+                Some(Err(e)) => return Err(e),
+                None => {
+                    if let Some(bin) = component_iter.next() {
```

Comment:
This seems inconsistent with the `while` section below in the number of tasks being added to the todo list.